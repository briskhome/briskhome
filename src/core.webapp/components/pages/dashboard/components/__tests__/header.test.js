import React from 'react';
import toJson from 'enzyme-to-json';
import { mount, render } from 'enzyme';

import { DashboardHeader } from '../../';

describe('core.webapp -> components -> pages -> dashboard -> header', () => {
  const RealDate = Date;
  const mockGetHours = jest.fn();
  global.Date = jest.fn(() => ({
    getHours: mockGetHours,
  }));
  global.Date.now = RealDate.now;

  const store = {
    dispatch: jest.fn(),
    getState: jest.fn(),
    subscribe: jest.fn(),
  };

  beforeEach(() => {
    store.getState.mockReturnValue({ user: { firstName: 'Example' } });
  });

  afterAll(() => {
    global.Date = RealDate;
  });

  it('renders in the morning', () => {
    mockGetHours.mockReturnValue(9);
    const component = mount(<DashboardHeader store={store} />);
    expect(toJson(render(component))).toMatchSnapshot();
  });

  it('renders in the afternoon', () => {
    mockGetHours.mockReturnValue(13);
    const component = mount(<DashboardHeader store={store} />);
    expect(toJson(render(component))).toMatchSnapshot();
  });

  it('renders in the evening', () => {
    mockGetHours.mockReturnValue(21);
    const component = mount(<DashboardHeader store={store} />);
    expect(toJson(render(component))).toMatchSnapshot();
  });

  it('renders in the night', () => {
    mockGetHours.mockReturnValue(3);
    const component = mount(<DashboardHeader store={store} />);
    expect(toJson(render(component))).toMatchSnapshot();
  });
});
