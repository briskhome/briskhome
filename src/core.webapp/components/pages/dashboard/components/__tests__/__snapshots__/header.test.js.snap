// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.webapp -> components -> pages -> dashboard -> header renders in the afternoon 1`] = `
<div
  class="dashboard-header"
>
  <h1
    class="briskhome-title briskhome-title_large dashboard-header__title"
  >
    Have a good afternoon, Example
  </h1>
  <div
    class="dashboard-header__status dashboard-header__status_good"
  >
    <p
      class="dashboard-header__status-text"
    >
      Everything is doing great. 
      <a
        class="dashboard-header__status-link"
        href="#"
      >
        See status.
      </a>
    </p>
  </div>
  <div
    class="dashboard-header__card-container"
  >
    <div
      class="dashboard-header__card-container-fade_left"
    />
    <a
      class="dashboard-card"
      href="#"
    >
      <span
        class="briskhome-icon briskhome-icon__add dashboard-card__icon"
      />
      <p
        class="dashboard-card__status"
      >
        Click here to add a new room to your house.
      </p>
    </a>
    <div
      class="dashboard-header__card-container-fade_right"
    />
  </div>
</div>
`;

exports[`core.webapp -> components -> pages -> dashboard -> header renders in the evening 1`] = `
<div
  class="dashboard-header"
>
  <h1
    class="briskhome-title briskhome-title_large dashboard-header__title"
  >
    Have a good evening, Example
  </h1>
  <div
    class="dashboard-header__status dashboard-header__status_good"
  >
    <p
      class="dashboard-header__status-text"
    >
      Everything is doing great. 
      <a
        class="dashboard-header__status-link"
        href="#"
      >
        See status.
      </a>
    </p>
  </div>
  <div
    class="dashboard-header__card-container"
  >
    <div
      class="dashboard-header__card-container-fade_left"
    />
    <a
      class="dashboard-card"
      href="#"
    >
      <span
        class="briskhome-icon briskhome-icon__add dashboard-card__icon"
      />
      <p
        class="dashboard-card__status"
      >
        Click here to add a new room to your house.
      </p>
    </a>
    <div
      class="dashboard-header__card-container-fade_right"
    />
  </div>
</div>
`;

exports[`core.webapp -> components -> pages -> dashboard -> header renders in the morning 1`] = `
<div
  class="dashboard-header"
>
  <h1
    class="briskhome-title briskhome-title_large dashboard-header__title"
  >
    Have a good morning, Example
  </h1>
  <div
    class="dashboard-header__status dashboard-header__status_good"
  >
    <p
      class="dashboard-header__status-text"
    >
      Everything is doing great. 
      <a
        class="dashboard-header__status-link"
        href="#"
      >
        See status.
      </a>
    </p>
  </div>
  <div
    class="dashboard-header__card-container"
  >
    <div
      class="dashboard-header__card-container-fade_left"
    />
    <a
      class="dashboard-card"
      href="#"
    >
      <span
        class="briskhome-icon briskhome-icon__add dashboard-card__icon"
      />
      <p
        class="dashboard-card__status"
      >
        Click here to add a new room to your house.
      </p>
    </a>
    <div
      class="dashboard-header__card-container-fade_right"
    />
  </div>
</div>
`;

exports[`core.webapp -> components -> pages -> dashboard -> header renders in the night 1`] = `
<div
  class="dashboard-header"
>
  <h1
    class="briskhome-title briskhome-title_large dashboard-header__title"
  >
    Have a good night, Example
  </h1>
  <div
    class="dashboard-header__status dashboard-header__status_good"
  >
    <p
      class="dashboard-header__status-text"
    >
      Everything is doing great. 
      <a
        class="dashboard-header__status-link"
        href="#"
      >
        See status.
      </a>
    </p>
  </div>
  <div
    class="dashboard-header__card-container"
  >
    <div
      class="dashboard-header__card-container-fade_left"
    />
    <a
      class="dashboard-card"
      href="#"
    >
      <span
        class="briskhome-icon briskhome-icon__add dashboard-card__icon"
      />
      <p
        class="dashboard-card__status"
      >
        Click here to add a new room to your house.
      </p>
    </a>
    <div
      class="dashboard-header__card-container-fade_right"
    />
  </div>
</div>
`;
